.github/workflows/release.yml

```diff
@@ -6,37 +6,107 @@ on:
     branches:
       - master
 
+env:
+  # Pin to v10.28.0, which (as of 2021-05-25) is the latest version with assets
+  AUTO_VERSION: v10.28.0
+
 jobs:
-  auto-release:
+  release-check:
     runs-on: ubuntu-latest
     if: "!contains(github.event.head_commit.message, 'ci skip') && !contains(github.event.head_commit.message, 'skip ci')"
     steps:
       - name: Checkout source
         uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
 
       - name: Download auto
         run: |
-          # Pin to v10.28.0, which (as of 2021-05-25) is the latest version
-          # with assets
-          auto_download_url="$(curl -fsSL https://api.github.com/repos/intuit/auto/releases/tags/v10.28.0 | jq -r '.assets[] | select(.name == "auto-linux.gz") | .browser_download_url')"
+          auto_download_url="$(curl -fsSL https://api.github.com/repos/intuit/auto/releases/tags/$AUTO_VERSION | jq -r '.assets[] | select(.name == "auto-linux.gz") | .browser_download_url')"
           wget -O- "$auto_download_url" | gunzip > ~/auto
           chmod a+x ~/auto
 
+      - name: Check whether a release is due
+        id: auto-version
+        run: |
+          version="$(~/auto version)"
+          echo "::set-output name=version::$version"
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  release:
+    runs-on: ubuntu-latest
+    needs: release-check
+    if: needs.release-check.outputs.auto-version != ''
```

Comment:
checking, possibly just for my own education, it is 
```
echo "::set-output name=version::$version"
```
above which is stored into `auto-version`. So isn't it guaranteed to always have non empty value for `auto-version` even if `$version` is empty (dunno what `auto` reports whenever no release is due`).?

Reply:
@yarikoptic The above line causes the value of `$version` (which is empty when no release is due) to be stored in `steps.auto-version.outputs.version`, which is then stored in `release-check.outputs.auto-version` by the `outputs:` block at the top of the job.